### PR TITLE
Feature/load multiple aggregates

### PR DESCRIPTION
**Pull request by lvonk**
Instead of `ids.map {|id|repository.load_aggregate(id)}` you can do `repository.load_aggregate(ids)`.

### CODE REVIEW

**Comment by lvonk** [comment]
Ping @erikrozendaal @bjpbakker

File: lib/sequent/core/aggregate_repository.rb [hunk 1]
```diff
@@ -60,12 +60,43 @@ def load_aggregate(aggregate_id, clazz = nil)
...
 
-        result
+        _aggregates = aggregates.values_at(*aggregate_ids).compact
+        _query_ids = aggregate_ids - _aggregates.map(&:id)
+
+        _aggregates += @event_store.load_events_for_aggregates(_query_ids).map do |stream, events|
+          raise AggregateNotFound.new(stream.aggregate_id) unless stream
+          aggregate_class = Class.const_get(stream.aggregate_type)
+          aggregate_class.load_from_history(stream, events)
+        end unless _query_ids.empty?
```
Comment (erikrozendaal) [inline lib/sequent/core/aggregate_repository.rb]: `EventStore#load_events_for_aggregates` already handles empty arrays, so no need for the `unless`.

File: lib/sequent/core/aggregate_repository.rb [hunk 2]
```diff
@@ -60,12 +60,43 @@ def load_aggregate(aggregate_id, clazz = nil)
...
+
+        _aggregates.map do |aggregate|
+          raise TypeError, "#{aggregate.class} is not a #{clazz}" if clazz && !(aggregate.class <= clazz)
+          aggregates[aggregate.id] = aggregate
```
Comment (erikrozendaal) [inline lib/sequent/core/aggregate_repository.rb]: First check all aggregates for type errors, then add to aggregates map?

File: lib/sequent/core/event_store.rb [hunk 3]
```diff
@@ -79,6 +79,29 @@ def load_events(aggregate_id)
         [stream.event_stream, events]
       end
 
+      def load_events_for_aggregates(aggregate_ids)
```
Comment (erikrozendaal) [inline lib/sequent/core/event_store.rb]: Simply delegate `load_events` to this method by wrapping `aggregate_id` in an array?

File: lib/sequent/core/aggregate_repository.rb [hunk 4]
```diff
@@ -60,12 +60,43 @@ def load_aggregate(aggregate_id, clazz = nil)
...
+      # Note: This will load all the aggregates in memory, so querying 100s of aggregates
+      # with 100s of events could cause memory issues.
+      #
+      # Only returns the aggregates that exists.
+      # If +clazz+ is given and one of the aggregates is not of the correct type
+      # a +TypeError+ is raised.
+      #
+      # +aggregate_ids+ The ids of the aggregates to be loaded
+      # +clazz+ Optional argument that checks if all aggregates are of type +clazz+
+      def load_aggregates(aggregate_ids, clazz = nil)
```
Comment (erikrozendaal) [inline lib/sequent/core/aggregate_repository.rb]: Simply delegate `load_aggregate` to this method by wrapping `aggregate_id` in an array?

**Pull request merged**